common/src/chain/transaction/transaction_v1.rs

```diff
@@ -7,7 +7,7 @@ use parity_scale_codec_derive::{Decode as DecodeDer, Encode as EncodeDer};
 
 use super::Transaction;
 
-#[derive(Debug, Clone, PartialEq, Eq, EncodeDer, DecodeDer)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, EncodeDer, DecodeDer)]
```

Comment:
Why would we need to have a defined order of transactions?

Reply:
I think it's a requirement imposed by the `Block` struct's `Ord`/`PartialOrd`

Reply:
@TheQuantumPhysicist `cached_inputs: BTreeMap<Id<Transaction>, TxMainChainIndex>` required `Ord` and `PartialOrd`  because of the requirements on the impl of `Ord` for `Id<common::chain::Transaction>`. 

And there is not an order trait despite the naming, it's Total Order that means we add possibility to compare objects.

Reply:
This was fixed in another commit in this PR. Resolving this.